bundles/org.openhab.binding.juicenet/src/main/resources/OH-INF/thing/juicenet-device.xml

```diff
@@ -0,0 +1,271 @@
+		<state readOnly="true" pattern="%.2f kW"/>
+	</channel-type>
+
+	<channel-type id="seconds_charging_type">
+		<item-type>Number:Time</item-type>
+		<label>Charging Time</label>
+		<description>Charging time since plug-in time. (s)</description>
+		<state readOnly="true" pattern="%.0f min"/>
+	</channel-type>
+
+	<channel-type id="wh_energy_at_plugin_type">
+		<item-type>Number:Energy</item-type>
+		<label>Energy at Plugin</label>
+		<description>Energy value at the plugging time. (Wh)</description>
+		<state readOnly="true" pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="wh_energy_to_add_type">
+		<item-type>Number:Energy</item-type>
+		<label>Energy to Add</label>
+		<description>Amount of energy to be added in current session. (Wh)</description>
+		<state pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="lifetime_wh_energy_type">
+		<item-type>Number:Energy</item-type>
+		<label>Lifetime Energy</label>
+		<description>Total energy poured to vehicles during lifetime. (Wh)</description>
+		<state readOnly="true" pattern="%,.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="lifetime_savings_type">
+		<item-type>Number</item-type>
+		<label>Lifetime Savings</label>
+		<description>EV driving saving during lifetime.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="gascost_type">
+		<item-type>Number</item-type>
+		<label>Gas Cost</label>
+		<description>Cost of gasoline used in savings calculations.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="mpg_type">
+		<item-type>Number</item-type>
+		<label>MPG</label>
+		<description>Miles per gallon used in savings calculations.</description>
+		<state readOnly="true" pattern="%.1f mpg"/>
+	</channel-type>
+
+	<channel-type id="ecost_type">
+		<item-type>Number</item-type>
+		<label>Utility Energy Cost</label>
+		<description>Cost of electricity from utility company. ($/kWh)</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="whpermile_type">
+		<item-type>Number:Power</item-type>
+		<label>Watts per mile</label>
+		<description>Watts per mile.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="car_description_type">
+		<item-type>String</item-type>
+		<label>Car Description</label>
+		<description>Car description of vehicle currently or last charged.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="car_battery_size_wh_type">
+		<item-type>Number:Energy</item-type>
+		<label>Car Battery Pack Size</label>
+		<description>Car battery pack size. (Wh)</description>
+		<state readOnly="true" pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="car_battery_range_m_type">
+		<item-type>Number:Length</item-type>
+		<label>Mileage Range</label>
+		<description>Car mileage range.</description>
+		<state readOnly="true" pattern="%,d miles"/>
```

Comment:
Why is the pattern hard coded to display miles?

Reply:
This one can be made to use UOM since there is a distance UOM defined.

Reply:
@Skinah - I'm resolving this comment on your behalf since this was fixed recently.